Reverse proxy transforms

ErikApption opened the pull request:
## Description
- this PR adds Yarp Transforms instead of a middleware to check access to proxied URL

## How Has This Been Tested?
- Locally tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Unit test (new or update to tests)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (change that would cause documentation to be updated)
- [ ] Configuration change (change that would affect the configuration of the application)

## Checklist:

- [X] PR is submitted to the correct branch `develop`.
- [X] Code follows the code style of this project.
- [ ] Changes are covered by Specflow tests and all new or existing tests are passing.
- [ ] Any new or updated translatable strings have been added to the localization files.
- [ ] Necessary and relevant documentation has been created or updated.
- [ ] There are either no changes to the application's configuration or the necessary changes in the infrastructure repository are included in a separate PR.

<!-- Link to the "infrastructure" repository PR here -->

ErikApption committed:
standarized all env tests into one single method

ErikApption committed:
added transforms with http context

ErikApption committed:
updated git ignore

ErikApption committed:
merged develop

ErikApption committed:
removed middleware and added role check

ErikApption committed:
re-organized service init

yjmrobert commented:
Would be nice to get some specflow tests on this to catch any string changes on the role constants

ErikApption commented:
the role constants in GetWorkspaceRole should have no impact - the output is only tested for null/not null. I was wondering how to cover this with specflow... if `HttpContext` can be instantiated in tests then some parts of the auth pipeline could be tested.

yjmrobert committed:
🧪 Added unit test to catch any changes to roles

yjmrobert committed:
🔧 Refactor reverse proxy service configuration

Moved the reverse proxy service setup to a dedicated configuration class. This enhances code maintainability and simplifies the Startup and ConfigureServices files by isolating the reverse proxy logic.
